packages/hardhat/contracts/PixelCanvas.sol

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract PixelCanvas is Ownable {
+    // Canvas Dimensions
+    uint256 public constant CANVAS_WIDTH = 64;
+    uint256 public constant CANVAS_HEIGHT = 64;
+
+    // Color Palette (Limited Options)
+    enum Color {
+        WHITE,
+        BLACK,
+        RED,
+        GREEN,
+        BLUE,
+        YELLOW,
+        PURPLE,
+        ORANGE
+    }
+
+    // Pixel Structure
+    struct Pixel {
+        address author;
+        Color color;
+        uint256 timestamp;
+    }
+
+    // 2D Mapping to store pixels
+    Pixel[CANVAS_WIDTH][CANVAS_HEIGHT] public canvas;
```

Comment:
Use a mapping here instead. That is more gas-efficient and flexible, storing the whole array from the beginning a is not the ideal solution. The key advantages of a mapping compared to a fixed-size array are:
- Lower Deployment Cost: There's no need to initialize the entire array, reducing upfront costs.
- Higher Gas Efficiency: It only stores modified pixels, making it ideal for canvases where most pixels remain unchanged

Consider what additional functions we might not need if we switch to using a mapping.